www/src/templates/template-blog-post.js

```diff
@@ -35,12 +35,19 @@ class BlogPostTemplate extends React.Component {
       </p>
     )
     let canonicalLink
+    let ogUrl
 
     if (post.frontmatter.canonicalLink) {
       canonicalLink = (
         <link rel="canonical" href={post.frontmatter.canonicalLink} />
       )
+      ogUrl = (
+        <meta property="og:url" content={post.frontmatter.canonicalLink} />
+      )
+    } else {
+      ogUrl = <meta property="og:url" content={href} />
```

Comment:
If there is no `canonicalLink` (i.e. this article isn't a crosspost of an article from another site), then we don't need to set `ogUrl` here since it'll be inherit the one in `SiteMetadata`. I think it's better to restructure like this:

```jsx
const { canonicalLink } = post.frontmatter

// In render
<Helmet>
  {canonicalLink && <link rel="canonical" href={canonicalLink} />
  {canonicalLink && <meta property="og:url" content={canonicalLink} />
  {/* other helmet stuff */}
</Helmet>
</Helmet>

Reply:
sometimes the canonicalLink is null. So the below would at lease return a value?
```javascript
{canonicalLink ? (
  <meta property="og:url" content={canonicalLink} />
) : (
  <meta property="og:url" content={href} />
)}
```